packages/async-flow/src/replay-membrane.js

```diff
@@ -201,6 +206,157 @@ export const makeReplayMembrane = (
     }
   };
 
+  // //////////////// Eventual Send ////////////////////////////////////////////
+
+  /**
+   * @param {PassableCap} hostTarget
+   * @param {string | undefined} optVerb
+   * @param {Passable[]} hostArgs
+   * @param {number} callIndex
+   * @param {VowKit} hostResultKit
+   * @param {Promise} guestReturnedP
+   * @returns {Outcome}
+   */
+  const performSend = (
+    hostTarget,
+    optVerb,
+    hostArgs,
+    callIndex,
+    hostResultKit,
+    guestReturnedP,
+  ) => {
+    const { vow, resolver } = hostResultKit;
+    try {
+      const hostPromise = optVerb
+        ? E(hostTarget)[optVerb](...hostArgs)
+        : E(hostTarget)(...hostArgs);
+      resolver.resolve(hostPromise); // TODO does this always work?
```

Comment:
Yeah thinking about this, if `hostTarget` is a local object for which call returns a promise, we will fail during an upgrade. Unlike the call case, we're not in a position to enforce this and fail unconditionally, unless we can somehow sniff the type of the `hostTarget` (if it has an eventual handler or not), and bypass the `E` call if it doesn't. If the `hostTarget` object does not return a promise, then the hostPromise will be fulfilled in the same crank, so it's safe.